mcarch/views/mods.py

```diff
@@ -14,20 +14,24 @@
 
 @modbp.route("/mods")
 def browse():
-    mod_query = Mod.query.filter_by(draft=False)
+    drafts_only = request.args.get('drafts', type=bool) and cur_user().has_role(roles.archivist)
+
+    mod_query = Mod.query.filter_by(draft=drafts_only)
```

Comment:
Something is wrong here. When I test on my system, no mods show up on the browse page regardless of whether `drafts` is True, False, or omitted. More investigation needed.

Reply:
When logged in?

Reply:
This is somewhat fixed now, but why check both `insecure_cur_user` and `cur_user`? This could still cause a problem if they disagree. For example, if a session expires server-side, `insecure_cur_user` will return a username, but `cur_user` will return `None` and then raise an error.

Maybe change to `cur_user() and cur_user().has_role...`?

Reply:
Wanted to avoid running the same database call twice in a row.
I could either:
a. change both to cur_user()
b. save cur_user in a variable while still short-circutting
c. update to 3.8 and use the walrus operator :smiling_imp: 

ps. does cur_user() cache?

Reply:
It doesn't cache, but it could be made to in the future pretty easily. Using a variable sounds reasonable.

Reply:
It probably will be cached per-request at some point in the future, since I think there are a few cases where it's called twice in a single request by totally different parts of the code, and having it cached would be the easiest way to optimize that. I wouldn't worry too much about it right now, though.